packages/synapse-react-client/src/lib/utils/hooks/useDetectSSOCode.ts

```diff
@@ -43,30 +43,26 @@ export default function useDetectSSOCode(
     onTwoFactorAuthRequired,
   } = opts
 
-  const [isLoading, setIsLoading] = useState(false)
+  const redirectURL = getRootURL()
+  // 'code' handling (from SSO) should be preformed on the root page, and then redirect to original route.
+  const fullUrl: URL = new URL(window.location.href)
+  // in test environment the searchParams isn't defined
+  const { searchParams } = fullUrl
+  const code = searchParams?.get('code')
+  const provider = searchParams?.get('provider')
+  // state is used during OAuth based Synapse account creation (it's the username)
+  const state = searchParams?.get('state')
+
+  const [isLoading, setIsLoading] = useState(!!(code && provider))
```

Comment:
Pulling this out of the effect will give us the correct loading state after the first render, which is easier to test.